c++/src/kj/memory.h

```diff
@@ -435,8 +435,13 @@ class HeapDisposer final: public Disposer {
   static const HeapDisposer instance;
 };
 
+#if defined(_MSC_VER) && !defined(__clang__)
+template <typename T>
+__declspec(selectany) const HeapDisposer<T> HeapDisposer<T>::instance = HeapDisposer<T>();
```

Comment:
My reading list was:
* https://developercommunity.visualstudio.com/t/already-defined-symbol-for-static-template-member/311664
* https://developercommunity.visualstudio.com/t/linker-failing-because-of-multiple-definitions-of/224597
* https://docs.microsoft.com/en-us/cpp/error-messages/tool-errors/linker-tools-error-lnk2005?view=msvc-160

Not sure how we prefer to gate these things, I could do this _only_ for MSVC 2017?

Reply:
Yeah, `_MSC_VER` should evaluate to some sort of version number which you can compare against. It looks values less than 1920  would correspond to MSVC 2017? https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros